internal/clients/http/sources/sources_client.go

```diff
@@ -183,20 +190,15 @@ func (c *sourcesClient) GetAuthentication(ctx context.Context, sourceId string)
 		return nil, fmt.Errorf("get source authentication call: %w", err)
 	}
 
-	// Filter authentications to include only auth where resource_type == "Application". We do this because
-	// Sources API currently does not provide a good server-side filtering.
-	auth, err := filterSourceAuthentications(*resp.JSON200.Data)
-	if err != nil {
-		at := make([]string, 0)
-		for _, auth := range *resp.JSON200.Data {
-			at = append(at, string(*auth.Authtype))
-		}
-		logger.Warn().Msgf("Sources did not return any Provisioning authentication for source(auth types): %s(%v)", sourceId, at)
-		return nil, err
+	if len(*resp.JSON200.Data) != 0 {
+		// TODO: What do we do if we have more than one authentication
+		auth := (*resp.JSON200.Data)[0]
+		authentication := clients.NewAuthenticationFromSourceAuthType(ctx, *auth.Username, string(*auth.Authtype), *auth.ResourceId)
+		return authentication, nil
```

Comment:
Honestly I would not worry about it here. Just check if len is > 1 and take the first element.